pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -79,7 +82,21 @@ int getPriority(BaseFilterOperator filterOperator) {
           return 3;
         }
         if (filterOperator instanceof ScanBasedFilterOperator) {
-          return 4;
+          DataSourceMetadata metadata = filterOperator.getDataSourceMetadata();
```

Comment:
Can we factor this out into a function? And also document the logic behind this heuristic, and the priority computation?

Reply:
Done.